enumerables.rb

```diff
@@ -0,0 +1,169 @@
+module Enumerable
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_each
+    x = 0
+    return to_enum unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    while x < list.length
+      yield(list[x])
+      x += 1
+    end
+    list
+  end
+
+  def my_each_with_index
+    x = 0
+    return to_enum unless block_given?
+
+    list = is_a?(Range) ? to_a : self
+    while x < list.length
+      yield(list[x], x)
+      x += 1
+    end
+    list
+  end
+
+  def my_select
+    result = []
+    return to_enum unless block_given?
+
+    my_each do |x|
+      result << x if yield(x)
+    end
+    result
+  end
+
+  def my_all?(input = nil)
+    my_each do |x|
+      return false if block_given? && !yield(x)
+
+      if !block_given? && input.nil?
+        return false unless x
+      elsif input
+        return false
+      end
+    end
+    true
+  end
+
+  def my_any?(input = nil)
+    my_each do |x|
+      return false if block_given? && yield(x)
+
+      if !block_given? && input.nil?
+        return true if x
+      elsif !block_given? && input
+        return true
+      end
+    end
+    false
+  end
```

Comment:
`my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- returns true if the block ever returns a value other than false or nil
```ruby
[1, 2, 3].my_any? { |num| num % 2 == 0 }
```
- when a class is passed as an argument returns true if at least one of the collection is a member of such class
```ruby
[1, 2, 3].my_any?(Integer) #should return true
[1, 'demo', false].my_any?(Integer) #should return true
['demo', false, nil].my_any?(Integer) #should return false
```
- when a Regex is passed as an argument returns false if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_any?(/t/)    #should return false.
%w[dog door rod blade].my_any?(/d/)   #should return true.
```
- when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
[1, 2, 3].my_any?(3)   #should return true
[1, 2, 2].my_any?(3)   #should return false
```